iceoryx_posh/test/moduletests/test_roudi.cpp

```diff
@@ -0,0 +1,545 @@
+/// @post describe what needs to be done in teardown()
+/// @note name of the Testfixture should match to the Class you want to test
+
+class RoudiClassTest : public RouDi
+{
+  public:
+    RoudiClassTest(iox::roudi::RouDiMemoryInterface& roudiMemoryInterface,
+                   iox::roudi::PortManager& portManager,
+                   iox::roudi::RouDi::RoudiStartupParameters roudiStartupParameters)
+        : RouDi(roudiMemoryInterface, portManager, roudiStartupParameters)
+    {
+    }
+
+    void startMQThreadTest()
+    {
+        this->startMQThread();
+    }
+
+    void shutDownTest()
+    {
+        this->shutdown();
+    }
+
+    void processMessageTest(const runtime::MqMessage& message,
+                            const iox::runtime::MqMessageType& cmd,
+                            const ProcessName_t& processName)
+    {
+        this->processMessage(message, cmd, processName);
+    }
+
+    const PortManager* getPortManager()
+    {
+        return m_portManager;
+    }
+};
+
+class RouDiBaseClass_test : public Test
+{
+  public:
+};
+
+TEST_F(RouDiBaseClass_test, ConstructorIsSuccessfull)
+{
+    std::unique_ptr<IceOryxRouDiComponents> m_roudiComponents =
+        std::unique_ptr<IceOryxRouDiComponents>(new IceOryxRouDiComponents(iox::RouDiConfig_t().setDefaults()));
+    std::unique_ptr<RoudiClassTest> m_roudiApp = std::unique_ptr<RoudiClassTest>(
+        new RoudiClassTest(m_roudiComponents->m_rouDiMemoryManager,
+                           m_roudiComponents->m_portManager,
+                           RouDi::RoudiStartupParameters{roudi::MonitoringMode::ON, false}));
+
+    EXPECT_NE(m_roudiApp, nullptr);
+
+    m_roudiApp->shutDownTest();
+}
+
+TEST_F(RouDiBaseClass_test, ConstructorThreadStartNotImmediateIsSuccessfull)
+{
+    std::unique_ptr<IceOryxRouDiComponents> m_roudiComponents =
+        std::unique_ptr<IceOryxRouDiComponents>(new IceOryxRouDiComponents(iox::RouDiConfig_t().setDefaults()));
+    std::unique_ptr<RoudiClassTest> m_roudiApp = std::unique_ptr<RoudiClassTest>(new RoudiClassTest(
+        m_roudiComponents->m_rouDiMemoryManager,
+        m_roudiComponents->m_portManager,
+        RouDi::RoudiStartupParameters{roudi::MonitoringMode::ON, false, RouDi::MQThreadStart::DEFER_START}));
+
+    m_roudiApp->startMQThreadTest();
+
+    EXPECT_DEATH(m_roudiApp->startMQThreadTest(), ".*");
+
+    m_roudiApp->shutDownTest();
+}
+
+TEST_F(RouDiBaseClass_test, ShutDownKillProcessInDestructorTrueIsSuccessfull)
+{
+    std::unique_ptr<IceOryxRouDiComponents> m_roudiComponents =
+        std::unique_ptr<IceOryxRouDiComponents>(new IceOryxRouDiComponents(iox::RouDiConfig_t().setDefaults()));
+    std::unique_ptr<RoudiClassTest> m_roudiApp = std::unique_ptr<RoudiClassTest>(
+        new RoudiClassTest(m_roudiComponents->m_rouDiMemoryManager,
+                           m_roudiComponents->m_portManager,
+                           RouDi::RoudiStartupParameters{roudi::MonitoringMode::ON, true}));
```

Comment:
this is repeated for almost every test and should be moved to `SetUp` method. The two tests with the ctor could be tests withoud a fixture, so just `TEST`, then it's easily possible